regolith/helpers/a_proposalhelper.py

```diff
@@ -0,0 +1,197 @@
+        rc = self.rc
+        rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        if not rc.database:
+            rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def db_updater(self):
+        gtx = self.gtx
+        rc = self.rc
+        if not rc.date:
+            now = dt.date.today()
+        else:
+            now = date_parser.parse(rc.date).date()
+        key = f"{str(now.year)[2:]}_{''.join(rc.name.casefold().split()).strip()}"
+
+        coll = self.gtx[rc.coll]
+        pdocl = list(filter(lambda doc: doc["_id"] == key, coll))
+        if len(pdocl) > 0:
+            raise RuntimeError(
+                "This entry appears to already exist in the collection")
+        else:
+            pdoc = {}
+        pdoc.update({'_id': key})
+        if rc.amount:
+            pdoc.update({'amount': rc.amount})
+        else:
+            pdoc.update({'amount': ''})
+        if rc.authors:
+            if isinstance(rc.authors, str):
+                pdoc.update({'authors': [rc.authors]})
+            else:
+                pdoc.update({'authors': rc.authors})
+        else:
+            pdoc.update({'authors': ['']})
+        if rc.begin_date:
+            pdoc.update({'begin_date': rc.begin_date})
+        else:
+            pdoc.update({'begin_date': ''})
+        cpp_info = {}
+        if rc.cppflag:
+            cpp_info['cppflag'] = rc.cppflag
+        if rc.other_agencies:
+            cpp_info['other_agencies_submitted'] = rc.other_agencies
+        if rc.institution:
+            cpp_info['institution'] = rc.institution
+        if rc.months_academic:
+             cpp_info['person_months_academic'] = rc.months_academic
+        if rc.months_summer:
+            cpp_info['person_months_summer'] =  rc.months_summer
+        if rc.scope:
+            cpp_info['project_scope'] = rc.scope
+        if cpp_info:
+            pdoc.update({'cpp_info':cpp_info})
+        if rc.currency:
+            pdoc.update({'currency': rc.currency})
+        else:
+            pdoc.update({'currency': 'USD'})
+        if rc.due_date:
+            pdoc.update({'due_date': rc.due_date})
+        else:
+            pdoc.update({'due_date': 'tbd'})
+        if rc.duration:
+            pdoc.update({'duration': rc.duration})
+        else:
+            pdoc.update({'duration': 'tbd'})
+        if rc.end_date:
+            pdoc.update({'end_date': rc.end_date})
+        else:
+            pdoc.update({'end_date': ''})
+        if rc.funder:
+            pdoc.update({'funder': rc.funder})
+        if rc.full:
+            pdoc.update({'full': rc.full})
+        if rc.notes:
+            pdoc.update({'notes': rc.notes})
+        if rc.pi:
+            pdoc.update({'pi': rc.pi})
```

Comment:
you seem to be handling this differently whether rc.pi exists or not, i.e., if pi is passed in you don't dereference it but if it is not then you do.  In general I think the best is to store ids in the database and only dereference them when we build something.

Reply:
done – I handled them differently because the`pi` help asks for the pi name, not the pi_id, so there is no need to retrieve if the user has specified a `pi`
but I will now simply change the code to display the `pi_id` in both cases, and change the `pi` help to "ID of principal investigator"

Reply:
perfect, that is the behavior I want.